src/olympia/landfill/views.py

```diff
@@ -0,0 +1,93 @@
+from django.db import connection
+from django.test.testcases import TransactionTestCase
+
+from olympia import amo
+from olympia.api.authentication import JWTKeyAuthentication
+from olympia.api.permissions import GroupPermission
+
+from .serializers import GenerateAddonsSerializer
+
+
+class GenerateAddons(APIView):
+    authentication_classes = [JWTKeyAuthentication]
+    permission_classes = [
+        IsAuthenticated,
+        GroupPermission(amo.permissions.ACCOUNTS_SUPER_CREATE)]
+
+    @waffle_switch('uitests-enable-landfill')
+    def post(self, request):
+        serializer = GenerateAddonsSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response({'errors': serializer.errors}, status=422)
+        serializer.create_addons()
+        cache.clear()
+        return Response({}, status=201)
+
+
+class DumpCurrentState(APIView):
+    authentication_classes = [JWTKeyAuthentication]
+    permission_classes = [
+        IsAuthenticated,
+        GroupPermission(amo.permissions.ACCOUNTS_SUPER_CREATE)]
+
+    @waffle_switch('uitests-enable-landfill')
+    def post(self, request):
+        state = datetime.datetime.utcnow().isoformat()
+        fname = '/tmp/{}.json'.format(state)
+        with open(fname, 'wb') as fobj:
+            fobj.write(connection.creation.serialize_db_to_string())
+
+        return Response({'state': state}, status=200)
+
+
+class RestoreCurrentState(APIView):
+    authentication_classes = [JWTKeyAuthentication]
+    permission_classes = [
+        IsAuthenticated,
+        GroupPermission(amo.permissions.ACCOUNTS_SUPER_CREATE)]
+
+    @waffle_switch('uitests-enable-landfill')
+    def post(self, request):
+        state = request.data.get('state')
+
+        fname = '/tmp/{}.json'.format(state)
+        if os.path.exists('/tmp/{}.json'.format(state)):
+            with open(fname, 'rb') as fobj:
+                data = fobj.read()
+
+        print('Found data to restore')
+
+        print('Truncate...')
+
+        # Alternate approach:
+        # * set "ui-tests-running" flag so that QA and others know not to touch
+        #   anything (e.g if on -dev, for docker... who cares...)
+        # * get next mysql sequence number
+        # * iterate through all apps and objects just like
+        #   `serialize_db_to_string` does
+        # * delete 'em all
+
+        # Truncate the whole database
+        databases = TransactionTestCase._databases_names(include_mirrors=False)
+        for db_name in databases:
+            # Flush the database
+            call_command('flush', verbosity=0, interactive=False,
+                         database=db_name, reset_sequences=False,
+                         allow_cascade=False,
+                         inhibit_post_migrate=False)
+
+        # print('Restore...')
+        # Restore database to previously known state
+        connection.creation.deserialize_db_from_string(data)
+
+        return Response({}, status=200)
```

Comment:
If we don't use the views for now we should just delete them until we figure out a way to fix them and make them work.